StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1211UnitTests.cs

```diff
@@ -0,0 +1,149 @@
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will not produce diagnostics for correctly ordered using directives in multiple namespaces.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestValidUsingDirectivesInMultipleNamespacesAsync()
+        {
+            const string testCode = @"namespace Foo
+{
+    using System;
+    using character = System.Char;
+    using \u0069nt = System.Int32;
+}
+
+namespace Bar
+{
+    using System;
+    using MemoryStream = System.IO.MemoryStream;
+    using Stream = System.IO.Stream;
+    using StringBuilder = System.Text.StringBuilder;
+    using StringWriter = System.IO.StringWriter;
+}
+
+namespace Spam
+{
+    using System;
+    using Character = System.Char;
+    using @int = System.Int32;
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will not produce diagnostics for correctly ordered using directives in the compilation unit.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestValidUsingDirectivesInCompilationUnitAsync()
+        {
+            const string testCode = @"using System;
+using character = System.Char;
+using \u0069nt = System.Int32;
+using StringBuilder = System.Text.StringBuilder;
+using StringWriter = System.IO.StringWriter;
+
+public class Foo
+{
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will produce the proper diagnostics when the using directives are ordered wrong.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestInvalidUsingDirectivesOrderingAsync()
+        {
+            const string testCode = @"namespace Foo
+{
+    using System;
+    using \u0069nt = System.Int32;
+    using character = System.Char;
+}
+
+namespace Bar
+{
+    using System;
+    using Stream = System.IO.Stream;
+    using StringBuilder = System.Text.StringBuilder;
+    using StringWriter = System.IO.StringWriter;
+    using MemoryStream = System.IO.MemoryStream;
+}
+
+namespace Spam
+{
+    using System;
+    using @int = System.Int32;
+    using Character = System.Char;
+}
+";
+
+            DiagnosticResult[] expectedDiagnostics =
+            {
+                this.CSharpDiagnostic().WithLocation(5, 5).WithArguments("character", "\u0069nt"),
```

Comment:
:bulb: The second argument can be written as just `"int"`; the escape sequence is misleading.